minimint/tests/TESTING.md

```diff
@@ -0,0 +1,85 @@
+# Integration Testing
+The integration tests allow developers to test the interactions between the Minimint Federation, LN gateway, Clients, Lightning and Bitcoin.
+
+## Writing tests
+Tests cases begin by initializing test fixtures with the number of federation nodes, the threshold that can misbehave, and the coin denomination tiers:
+
+```rust
+let (fed, user, bitcoin, gateway, ln) = fixtures(2, 0, &[sats(100), sats(1000)]).await;
+```
+
+Initialization will spawn API and HBBFT consensus threads for federation nodes starting at port `4000` then give you access to the following:
+- `fed`- control and inspect federation nodes and consensus
+- `user`- calls functions in the user client API to simulate minimint users
+- `bitcoin`- manipulate the shared Bitcoin network
+- `gateway`- calls functions in the gateway client API to simulate a gateway node
+- `lightning`- manipulate the gateway LN node and another connected LN node
+
+Calling functions on the clients can send requests to the federation's API and add new proposals to consensus:
+```rust
+user.client.peg_in(proof, tx, rng()).await.unwrap();
+```
+In order to simulate consensus we have to tell the federation how many epochs to run:
+```rust
+fed.run_consensus_epochs(2).await;
+```
+Note that because HBBFT and consensus processing are concurrent you must always add an additional epoch to consume a `ConsensusOutcome` before any existing `ConsensusItems` will be processed.
+
+## Running tests
+Tests run by default with fake Lightning and Bitcoin services for fast concurrent testing that succeeds in any environment, but can also be run against real services.
+
+To run the tests in parallel against fake versions of Lightning and Bitcoin:
+```shell
+export MINIMINT_TEST_REAL=0
```

Comment:
Isn't `MINIMINT_TEST_REAL=1` the default? Could it be omitted?

Reply:
Actually `MINIMINT_TEST_REAL=0` is the default.  It can be omitted, but I just included it in case they already had it set to `1` in their environment.

Reply:
That's what I meant lol